example/face_fields/face_fields_example.cpp

```diff
@@ -23,94 +22,105 @@
-      auto& summed = pmb->real_container.Get("c.c.interpolated_sum");
-      for (int k = ks; k <= ke; k++) {
-        for (int j = js; j <= je; j++) {
-          for (int i = is; i <= ie; i++) {
-            rank_sum += summed(k,j,i);
-          }
+DriverStatus FaceFieldExample::Execute() {
+  DriverUtils::ConstructAndExecuteBlockTasks<>(this);
+
+  // post-evolution analysis
+  Real rank_sum = 0.0;
+  MeshBlock *pmb = pmesh->pblock;
+  while (pmb != nullptr) {
+    int is = pmb->is;
+    int js = pmb->js;
+    int ks = pmb->ks;
+    int ie = pmb->ie;
+    int je = pmb->je;
+    int ke = pmb->ke;
+    auto &summed = pmb->real_container.Get("c.c.interpolated_sum");
+    for (int k = ks; k <= ke; k++) {
+      for (int j = js; j <= je; j++) {
+        for (int i = is; i <= ie; i++) {
+          rank_sum += summed(k, j, i);
         }
       }
-      pmb = pmb->next;
     }
+    pmb = pmb->next;
+  }
 #ifdef MPI_PARALLEL
-    Real global_sum;
-    MPI_Reduce(&rank_sum, &global_sum, 1, MPI_ATHENA_REAL, MPI_SUM, 0, MPI_COMM_WORLD);
+  Real global_sum;
+  MPI_Reduce(&rank_sum, &global_sum, 1, MPI_ATHENA_REAL, MPI_SUM, 0, MPI_COMM_WORLD);
 #else
-    Real global_sum = rank_sum;
+  Real global_sum = rank_sum;
 #endif
-    if (parthenon::Globals::my_rank == 0) {
-      std::cout << "\n\n"
-                << "Sum of all interpolated vars = " << global_sum << "\n"
-                << "It should be 0.0\n"
-                << std::endl;
-    }
-
-    pmesh->mbcnt = pmesh->nbtotal;
-    return DriverStatus::complete;
+  if (parthenon::Globals::my_rank == 0) {
+    std::cout << "\n\n"
+              << "Sum of all interpolated vars = " << global_sum << "\n"
+              << "It should be 0.0\n"
+              << std::endl;
   }
 
-  TaskList FaceFieldExample::MakeTaskList(MeshBlock *pmb) {
-    // make a task list for this mesh block
-    TaskList tl;
-    TaskID none(0);
-
-    auto fill_faces = tl.AddTask<BlockTask>(FaceFields::fill_faces, none, pmb);
+  pmesh->mbcnt = pmesh->nbtotal;
+  return DriverStatus::complete;
+}
 
-    auto interpolate = tl.AddTask<BlockTask>([](MeshBlock* pmb)->TaskStatus {
-        Container<Real>& rc = pmb->real_container;
-        int is = pmb->is; int js = pmb->js; int ks = pmb->ks;
-        int ie = pmb->ie; int je = pmb->je; int ke = pmb->ke;
-        auto& face = rc.GetFace("f.f.face_averaged_value");
-        auto& cell = rc.Get("c.c.interpolated_value");
+TaskList FaceFieldExample::MakeTaskList(MeshBlock *pmb) {
+  // make a task list for this mesh block
+  TaskList tl;
+  TaskID none(0);
+
+  auto fill_faces = tl.AddTask<BlockTask>(FaceFields::fill_faces, none, pmb);
+
+  auto interpolate = tl.AddTask<BlockTask>(
+      [](MeshBlock *pmb) -> TaskStatus {
+        Container<Real> &rc = pmb->real_container;
+        int is = pmb->is;
+        int js = pmb->js;
+        int ks = pmb->ks;
+        int ie = pmb->ie;
+        int je = pmb->je;
+        int ke = pmb->ke;
```

Comment:
this is cumbersome, but it will be solved when we finally have index shape structs in